### PR TITLE
Create `BasePipeline`

### LINKED ISSUE #562: Create `BasePipeline`
### 🚀 Feature Request

Create base class for all the pipelines

### Motivation

Now pipelines share the common logic however they don't have the base class to contain it

### Proposal

1. Create class `BasePipeline` (already exists):
- inherit it from `AbstractPipeline`, `BaseMixin`
- create abstract method `_forecast`(will be used inside the `forecast`): 
`def _forecast() -> TSDataset`
- create method `_forecast_prediction_interval`:
    - `def _forecast_prediction_interval(forecasts: TSDataset)` 
    - use implementation from `Pipeline`
    - now it should use forecasts from the outside instead of getting it [here](https://github.com/tinkoff-ai/etna/blob/1e1b6e368ee2fbd51dac59715255de9900e389eb/etna/pipeline/pipeline.py#L124)
- create method `_validate_quantiles`:
     - use current implementation
- implement method `forecast`:
   - should validate quantiles using `_validate_quantiles`
   - call `_forecast` inside and handle the prediction intervals logic using `_forecast_prediction_interval`
- implement method `backtest`:
    - use implementation from `Pipeline`
    - move all the necessary methods from `Pipeline` to this class
2. For `Pipeline`:
- inherit it from `BasePipeline`
- override method `forecast`:
    - if prediction_intervals=True and `isinstance(self.model, PerSegmentPredictionIntervalModel)`, then call the forecast of the model
    - else, call the `forecast` of the super class
- implement method `_forecast`, which return the predictions of the model without prediction intervals 
3. For all the other "pipelines":
- inherit them from `BasePipeline`
- rename method `forecast` to `_forecast`

### Test cases

_No response_

### Alternatives

_No response_

### Additional context

Blocked by #560 

### Checklist

- [ ] I discussed this issue with ETNA Team

### PR DESCRIPTION
**Pull request by alex-hse-repository**
**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

## Before submitting (must do checklist)

- [x] Did you read the [contribution guide](https://github.com/tinkoff-ai/etna/blob/master/CONTRIBUTING.md)?
- [x] Did you update the docs? We use Numpy format for all the methods and classes.
- [ ] Did you write any new necessary tests?
- [x] Did you update the [CHANGELOG](https://github.com/tinkoff-ai/etna/blob/master/CHANGELOG.md)?

## Type of Change

- [ ] Examples / docs / tutorials / contributors update
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Improvement (non-breaking change which improves an existing feature)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Closing issues

closes #562

### CODE REVIEW

File: etna/pipeline/__init__.py [hunk 1]
```diff
@@ -1,2 +1,3 @@
 from etna.pipeline.autoregressive_pipeline import AutoRegressivePipeline
+from etna.pipeline.base import BasePipeline
```
Comment (iKintosh) [inline etna/pipeline/__init__.py:2]: I don't think we need it, because it is not intended for users.

File: etna/pipeline/base.py [hunk 2]
```diff
@@ -124,13 +132,279 @@ def _validate_quantiles(quantiles: Sequence[float]) -> Sequence[float]:
                 raise ValueError("Quantile should be a number from (0,1).")
         return quantiles
 
-    def check_support_prediction_interval(self, prediction_interval_option: bool = False):
-        """Check if pipeline supports prediction intervals, if not, warns a user.
+    @abstractmethod
+    def _forecast(self) -> TSDataset:
+        """Make predictions."""
+        pass
```
Comment (iKintosh) [inline etna/pipeline/base.py:138]: It should have be
```suggestion
        raise NotImplementedError
```

File: etna/ensembles/voting_ensemble.py [hunk 3]
```diff
@@ -116,20 +116,12 @@ def _vote(self, forecasts: List[TSDataset]) -> TSDataset:
...
-        -------
-        TSDataset:
-            dataset with forecasts
+    def _forecast(self) -> TSDataset:
+        """Make predictions.
+        Compute weighted average of pipelines' forecasts
         """
-        self.check_support_prediction_interval(prediction_interval)
+        if self.ts is None:
+            raise ValueError("Something went wrong, ts is None inside the _forecast!")
```
Comment (iKintosh) [inline etna/ensembles/voting_ensemble.py:124]: We do not need to specify ` inside the _forecast!` bit, because we will have stack trace.
```suggestion
            raise ValueError("Something went wrong, ts is None")
```

File: etna/pipeline/autoregressive_pipeline.py [hunk 4]
```diff
@@ -81,12 +80,13 @@ def fit(self, ts: TSDataset) -> Pipeline:
 
         Returns
         -------
-        Pipeline:
+        self:
             Fitted Pipeline instance
         """
-        self.ts = deepcopy(ts)
+        self.ts = ts
```
Comment (iKintosh) [inline etna/pipeline/autoregressive_pipeline.py:86]: Why did you remove deepcopy?

File: etna/pipeline/autoregressive_pipeline.py [hunk 5]
```diff
@@ -81,12 +80,13 @@ def fit(self, ts: TSDataset) -> Pipeline:
...
         ts.fit_transform(self.transforms)
         self.model.fit(ts)
+        self.ts.inverse_transform()
```
Comment (iKintosh) [inline etna/pipeline/autoregressive_pipeline.py:89]: 👍

**Review by iKintosh — CHANGES_REQUESTED** [review]
I think it needs some fixes.

File: etna/pipeline/autoregressive_pipeline.py (hunk 4 above)
Comment (alex-hse-repository) [reply etna/pipeline/autoregressive_pipeline.py:86]: We don't have it in the Pipeline

File: etna/pipeline/base.py (hunk 2 above)
Comment (alex-hse-repository) [reply etna/pipeline/base.py:138]: We should change it every where then, I will create an issue

File: etna/ensembles/voting_ensemble.py (hunk 3 above)
Comment (alex-hse-repository) [reply etna/ensembles/voting_ensemble.py:124]: Done

**Review by iKintosh — APPROVED** [review]
👍
